stdlib/ballerina-builtin/src/main/ballerina/ballerina/io/natives.bal

```diff
@@ -139,6 +155,13 @@ public native function openFile (string path, string accessMode) (ByteChannel);
 @Return {value:"Socket which will allow to communicate with a remote server"}
 public native function openSocket (string host, int port, SocketProperties options) (Socket);
 
+@Description {value:"Open a secure socket connection with a remote server"}
+@Param {value:"host: Remote server domain/IP"}
+@Param {value:"port: Remote server port"}
+@Param {value:"options: Connection stream that bridge the client and the server"}
+@Return {value:"Socket which will allow to communicate with a remote server"}
+public native function openSecureSocket (string host, int port, SocketProperties options) (Socket);
```

Comment:
Can't we merge these two native functions? Or is it by design?

public native function openSocket (string host, int port, SocketProperties options) (Socket);

public native function openSecureSocket (string host, int port, SocketProperties options) (Socket);

Reply:
These two functions perform two different use cases. openSocket function simply opens a TCP connection with the remote server. So internal implementation is different. Secure socket implementation does little more validation to make sure the secure communication. Having two different methods will simplify the implementation.

Reply:
+1